addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -570,40 +628,165 @@ private boolean performCloseTagAttack(HtmlContext context, HttpMessage msg, Stri
         return false;
     }
 
-    private boolean performScriptAttack(HtmlContext context, HttpMessage msg, String param) {
-        List<HtmlContext> contexts2 =
-                performAttack(
-                        msg,
-                        param,
-                        context.getSurroundingQuote()
-                                + ";alert(1);"
-                                + context.getSurroundingQuote(),
-                        context,
-                        0);
-        if (contexts2 == null) {
+    private boolean performScriptAttack(HtmlContext context, String param) throws IOException {
+
+        String target = "eyeCatcher";
+
+        // resend request with eye catcher ideal for js
+        HttpMessage msg = getNewMsg();
+        setParameter(msg, param, target);
+        try {
+            sendAndReceive(msg);
+        } catch (URIException e) {
+            log.debug("Failed to send HTTP message, cause: {}", e.getMessage());
+            return false;
+        } catch (InvalidRedirectLocationException | UnknownHostException e) {
             return false;
         }
-        for (HtmlContext ctx : contexts2) {
-            if (context.getSurroundingQuote().isEmpty() || !ctx.getSurroundingQuote().isEmpty()) {
-                // Yep, its vulnerable
-                newAlert()
-                        .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                        .setParam(param)
-                        .setAttack(ctx.getTarget())
-                        .setEvidence(ctx.getTarget())
-                        .setMessage(ctx.getMsg())
-                        .raise();
-                return true;
-            }
+
+        // get context
+        ParserAPI parserApi = new ParserAPI(msg);
+        if (!parserApi.parseScript()) {
+            return false;
         }
-        return false;
+
+        String targetContext = parserApi.getContext(target);
+
+        // 1. direct attacks - context unaware
+        // simple direct attack in double quotes
+        String attack = "\";alert(1);//";
+        Boolean alertStatus = performAttackForScript(msg, param, attack);
```

Comment:
As pointed out by LGTM this should be 
```suggestion
        boolean alertStatus = performAttackForScript(msg, param, attack);
```

Reply:
I have updated :)